### PR TITLE
Add ARM64EC Build to CI Pipeline

### PR DESCRIPTION
**Pull request by sayanshaw24**
### Draft PR to check 'arm64ec' build on CI pipeline

Currently having issues building due to incompatibility of RelWithDebInfo config with 'arm64ec' platform: https://aiinfra.visualstudio.com/Lotus/_build/results?buildId=268301&view=logs&j=707d0f1b-a492-5d8f-3b99-86b6f845f8d2&t=642068fe-aa83-50f1-7a47-146324df7eb5

Created a new batch file `build-hardware-agnostic.bat` since `build.bat` hardcodes x64 platform and `build.sh` can't be run easily on Windows.

### CODE REVIEW

File: .pipelines/ci.yml [hunk 1]
```diff
@@ -274,6 +274,103 @@ jobs:
           ctest -C RelWithDebInfo --output-on-failure
         displayName: Run C++ native tests
 
+  #############
+  # ARM64EC
+  #############
+
+  - job: ARM64ECBuild
+    pool:
+      vmImage: 'windows-2019'
```
Comment (wenbingl) [inline .pipelines/ci.yml:283]: try this latest vm-image: https://learn.microsoft.com/en-us/azure/devops/pipelines/agents/hosted?view=azure-devops&tabs=yaml

Comment (sayanshaw24) [reply .pipelines/ci.yml:283]: sounds good, thanks!

File: .pipelines/ci.yml [hunk 2]
```diff
@@ -274,6 +274,103 @@ jobs:
...
+          destinationFolder: '$(Build.SourcesDirectory)'
+          cleanDestinationFolder: false
+          overwriteExistingFiles: true
+        displayName: Unpack ONNXRuntime package.
+
+      # Update to latest version of VS 2022
+      - script: |
+           echo "$(Build.SourcesDirectory)\vs2022"
+           echo "$(Build.SourcesDirectory)\vs2022\vs_enterprise.exe"
+           echo "%ProgramFiles%\Microsoft Visual Studio\2022\Enterprise"
```
Comment (wenbingl) [inline .pipelines/ci.yml:318]: Don't hardcode the VS 2022 path? use the vswhere in the following.

File: .pipelines/ci.yml [hunk 3]
```diff
@@ -274,6 +274,103 @@ jobs:
...
+  # ARM64EC
+  #############
+
+  - job: ARM64ECBuild
+    pool:
+      vmImage: 'windows-2019'
+    strategy:
+      matrix:
+        ort-1121:
+          ort.version: '1.12.1'
```
Comment (wenbingl) [inline .pipelines/ci.yml:287]: Only keep the latest ORT 1.13.x

**Pull request closed without merging**